### ISSUE TITLE
Error messaging for "no geometry available" could be more useful

### ISSUE THREAD

**Issue by vielmetti** [censusreporter/censusreporter#127]
The use case:

I have a series of data sets which include zip codes. I'm trying to generate maps based on those zip codes. I know that the census data is ZCTAs, that's fine. 

What is tricky so far is that some of the zip codes - the ones that serve post office boxes only - don't have a geometry associated with them. So when I query like so

```shell
curl -s 'http://api.censusreporter.org/1.0/geo/show/tiger2013?geo_ids='86000US48106
{
  "error": "The ACS 2013 5-year release doesn't include GeoID(s) 86000US48106."
}
```

which is correct. However if I request geo_ids from a series of ZIPs, hoping for the best, I get

```
curl -s 'http://api.censusreporter.org/1.0/geo/show/tiger2013?geo_ids='86000US48103,86000US48104,86000US48105,86000US48106,86000US48107,86000US48108,86000US48109
{
  "error": "The ACS 2013 5-year release doesn't include GeoID(s) 86000US48107,86000US48106."
}
```

and so if I want to correct my mistake I need to parse the error message and resubmit.

The request then runs something like one of these:

* return an error message in a more easily parseable format (or document this format);
* return the geometry for the geo-ids that it can parse, ignoring errors, if so directed;
* return the geo-ids that it *can* parse, as well as the ones it can't parse.
* put up with me querying a bunch of zips to see if they exist

I'm OK with the last of these

**Comment by vielmetti** [comment]
Closing this - I figured out how to make use of the error message in a repeatable way (so long as it doesn't change I'm OK for now).

**Comment by JoeGermuska** [comment]
that seems like a dangerous bet ;-) and I hate this error message. See also #110. 

The main problem is that it's hard enough to get community volunteers to help with code on the stuff that's actually in *this* repo. Setting up people to develop against[censusreporter/census-api](https://github.com/censusreporter/census-api) is a lot trickier because of the data setup.

**Comment by vielmetti** [comment]
The code in question is at https://github.com/censusreporter/census-api/blob/2c302827ee9ea90252f4acfcdd55e591b4a5eb49/census_extractomatic/api.py#L2122 and following, if I read it right.

**Comment by JoeGermuska** [comment]
Yes. At the end of the method, if there are IDs you requested which are not present, that's considered an error condition.

It's probably the case that the judgement of error should be left to clients. Since the API calls return javascript objects, we could add a new key indicating the details of what was not found, etc to make it easier for clients to make that judgment.

So, steps: 
* change the return value of  [expand_geoids(…)](https://github.com/censusreporter/census-api/blob/2c302827ee9ea90252f4acfcdd55e591b4a5eb49/census_extractomatic/api.py#L2122) so that callers can readily know which IDs in the request are not included in the response.
* update the methods which use it (as appropriate):
  * [`/1.0/data/show/<acs>`](https://github.com/censusreporter/census-api/blob/2c302827ee9ea90252f4acfcdd55e591b4a5eb49/census_extractomatic/api.py#L2171) -- response `json` (our own format)
  * [`/1.0/geo/show/tiger2013`](https://github.com/censusreporter/census-api/blob/2c302827ee9ea90252f4acfcdd55e591b4a5eb49/census_extractomatic/api.py#L1598) --   * [`/1.0/data/download/<acs>`](https://github.com/censusreporter/census-api/blob/2c302827ee9ea90252f4acfcdd55e591b4a5eb49/census_extractomatic/api.py#L2323) -- response `zipfile`
response: `geojson`
* document the changes.

We could probably use a comparable solution for the first two. It's harder to figure out where to put the information in the ZIPFile so that a person will see it, but maybe it goes in the metadata? That whole function (`download_specified_data(...)`) could really do with some refactoring!
